q.js

```diff
@@ -345,6 +346,14 @@ function isObject(value) {
     return value === Object(value);
 }
 
+// Pad an array value with nulls to the specified length.  Longer arrays are left alone.
+function padArrayTo(arry,len) {
+    if ( arry.length < len ) {
+        arry = array_concat(arry, new Array(len - arry.length));
```

Comment:
This can be accomplished without allocations `array.length = Math.min(array.length, length)`